Watchman.Discord/Areas/Protection/Controllers/MuteUserController.cs

```diff
@@ -31,27 +36,28 @@ public MuteUserController(MutingService mutingService, UnmutingService unmutingS
             this._mutingHelper = mutingHelper;
         }
 
-        [DiscordCommand("mute")]
         [AdminCommand]
-        public async Task MuteUser(DiscordRequest request, Contexts contexts)
+        public async Task MuteUser(MuteCommand command, Contexts contexts)
         {
-            var requestParser = new MuteRequestParser(request, this._usersService, contexts);
-            var userToMute = requestParser.GetUser();
-            if (userToMute.Id == this._usersService.GetBot().Id)
+            var userToMute = await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            if (userToMute == null)
             {
-                throw new UserDidntMentionAnyUserException();
+                throw new UserNotFoundException($"<@!{command.User}>");  // todo: change to command.User.GetMention()
```

Comment:
co daje wykrzyknik po `@`?
`$"<@!{command.User}>"`

Reply:
przeważnie w takiej formie są wzmianki, w zasadzie nigdy nie widziałem aby było inaczej.

Reply:
z resztą to i tak jest tymczasowo, później to będzie zmienione (spójrz na komentarz w kodzie)